### PR TITLE
ambisense: set operation mode and quick veto

### PR DESCRIPTION
**Pull request by thomasgermain**
Here it is.

- Creating `AmbisenseRoomOperationMode`, I think it is a good idea to have, like other operation mode enum.
- Creating `AmbisenseDevice`, I don't know if ti is useful, but I think it's cleaner to have it
- Adding `system_id` in `AmbisenseRoom`
- Some tests for new API calls

Still in draft, because I would like to test it a bit more (either real API calls or unit test) and have some feedback from you :)

### CODE REVIEW

**Comment by signalkraft** [comment]
Looks great!

I removed one unnecessary `from_api` call (those are only needed, if you need custom init logic) and added one very simple model testcase to check that devices are correctly initialized.

For my own setup, I use `python3 -m myPyllant.tests.run_function_test` to do real API calls and check that everything works. It should only write the current values to the API, so your heating system shouldn't be affected. Feel free to add some ambisense calls to that and let me know when everything works, so I can merge.

**Comment by signalkraft** [comment]
I added a PR for the HA component here: https://github.com/signalkraft/mypyllant-component/pull/138

Would appreciate your feedback for that one!

File: src/myPyllant/api.py [hunk 1]
```diff
@@ -1178,3 +1180,86 @@ async def get_ambisense_rooms(self, system: System | str) -> list[dict]:
...
+        Cancels a previously set quick veto in a room
+
+        Parameters:
+            room: The target room
+        """
+        url = f"{await self.get_api_base()}/api/v1/ambisense/facilities/{room.system_id}/rooms/{room.room_index}/configuration/quick-veto"
+
+        await self.aiohttp_session.delete(url, headers=self.get_authorized_headers())
+        room.room_configuration.quick_veto_end_time = None
+        # TODO not sure what to do with room.room_configuration.temperature_setpoint
```
Comment (thomasgermain) [inline src/myPyllant/api.py:1264]: To me, this is the last open point, I don't know exactly what we can do ?

I would not touch the `temperature_setpoint` and let it be overridden by the next API call, what do you think ?

**Comment by thomasgermain** [comment]
It looks good to me, do you want something more for this PR ? Maybe wait for more test in the HA component itself ?

File: src/myPyllant/api.py (hunk 1 above)
Comment (signalkraft) [reply src/myPyllant/api.py:1264]: Yeah, that makes sense! This updates on model fields are only done to give a quicker feedback in the HA component - but when there's no defined value to go to, I also just left them for the next API update.

**Comment by signalkraft** [comment]
I'll merge & tag it now, do a pre-release in the component, and test some more myself. The myVAILLANT API is a moving target, and lot's of people have expressed interest in Ambisense. If anything breaks for anyone, there's plenty of older versions to pick from on PyPI.

**Pull request merged**